openedx-unsupported/configuration
# Adds variables to the analytics_api master config YAML file

Adds the following variables to the generated `/edx/etc/analytics_api.yml` file, allowing them to be configurable during provisioning:
- `DATE_FORMAT` and `DATETIME_FORMAT`: used to format dates and timestamps returned by the Analytics REST API.
- `DEFAULT_PAGE_SIZE` and `MAX_PAGE_SIZE`: used to paginate (currently only) Learner List API results.  However, it's likely that other paginated API results will want to use them too.

This change helps facility the use of the Analytics Learner API to generate downloadable reports tailored to customer needs.

**Discussions**: This change is related to [edx-analytics-data-api PR #128](https://github.com/edx/edx-analytics-data-api/pull/128), which has been discussed extensively with the Analytics team.

**Dependencies**: None

**Sandbox URL**: TBD - sandbox is being provisioned.

**Partner information**: 3rd party-hosted open edX instance

**Testing instructions**:
1. See [edx-analytics-data-api PR #128](https://github.com/edx/edx-analytics-data-api/pull/128) for testing setup instructions.
2. Checkout this PR branch on your vagrant instance under `/edx/app/edx_ansible`
3. Create an extra-vars.yml file containing some overrides for the variables in this PR, and any others you want to preserve, e.g.
   
   ``` yaml
   ANALYTICS_API_DATE_FORMAT: "date goes here: %Y-%m-%d"
   ANALYTICS_API_DATETIME_FORMAT: iso-8601
   ANALYTICS_API_DEFAULT_PAGE_SIZE: 1
   ANALYTICS_API_MAX_PAGE_SIZE: 3
   
   ANALYTICS_API_ELASTICSEARCH_LEARNERS_HOST: localhost:9223
   ANALYTICS_API_ELASTICSEARCH_LEARNERS_INDEX: roster
   ```
4. Re-run the updated `analytics_api` role:
   
   ```
   ssh vagrant
   source /edx/app/edx_ansible/venvs/edx_ansible/bin/activate
   cd /edx/app/edx_ansible/edx_ansible/playbooks
   ansible-playbook -i localhost, -c local run_role.yml -e role=analytics_api --extra-vars=@/path/to/extra-vars.yml
   ```
5. Note changes written to `/edx/etc/analytics_api.yml`.
6. Confirm the changes using the [Learner List API](http://localhost:8100/docs/#!/api/Learner_List):
   - Check `DATE*_FORMAT` changes by looking at the `last_updated` date field.
   - Check the `*_PAGE_SIZE` changes counting the number of returned records.

**Reviewers**
